OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/ReachableMethodAnalysis.scala

```diff
@@ -65,47 +55,41 @@ trait ReachableMethodAnalysis extends FPCFAnalysis with TypeConsumerAnalysis {
             return processMethodWithoutBody(callersEOptP);
 
         val tacEP = propertyStore(method, TACAI.key)
-
         if (tacEP.hasUBP && tacEP.ub.tac.isDefined) {
-            processMethod(callersEOptP, null, tacEP.asEPS)
+            processMethod(callersEOptP, NoCallers, Some(tacEP.asEPS))
         } else {
             InterimPartialResult(Set(tacEP), continuationForTAC(declaredMethod))
         }
     }
 
-    protected val processesMethodsWithoutBody = false
-
     protected def processMethodWithoutBody(
         eOptP: EOptionP[DeclaredMethod, Callers]
-    ): PropertyComputationResult = {
-        if (processesMethodsWithoutBody) {
-            processMethod(eOptP, null, null)
-        } else
-            NoResult
-    }
+    ): PropertyComputationResult = processMethod(eOptP, NoCallers, None)
 
     private[this] def processMethod(
-        eOptP: EOptionP[DeclaredMethod, Callers],
-        seen:  Callers,
-        tacEP: EPS[Method, TACAI]
+        eOptP:      EOptionP[DeclaredMethod, Callers],
+        oldCallers: Callers,
+        tacEPOpt:   Option[EPS[Method, TACAI]]
     ): ProperPropertyComputationResult = {
+        val newCallers = if (eOptP.hasUBP) eOptP.ub else NoCallers
```

Comment:
Wouldn't it be more sensible to turn this into an if, only executing the `forNewCalleeContexts` if there is a UBP?

Reply:
Since we need the `newCallers` for constructing a continuation and `forNewCalleeContexts`  does not do anything when `NoCallers` is given, I view this as reasonable and less verbose as well.